WaterLog_Backend/Procedures.cs

```diff
@@ -227,5 +228,84 @@ private string getSegmentStatus(int segmentId)
             var entry = list.Where(inlist => inlist.SegmentsId == segmentId).Last();
             return entry.EventType;
         }
+
+        //Calculates the data points of the wastage based on period
+        public DataPoints<DateTime,double> CalculatePeriodWastage(Period timeframe)
+        {
+            switch (timeframe)
+            {
+                case Period.Daily:
+                    return CalculateDailyWastage(_db.SegmentEvents.Where(a => a.EventType == "leak" && a.TimeStamp.Month == DateTime.Now.Month && a.TimeStamp.Day == DateTime.Now.Day && a.TimeStamp.Year == DateTime.Now.Year).GroupBy(b => b.TimeStamp.Hour).ToList());
```

Comment:
structure your returns in a readable manner 'where codition and groupby on its own lines  , rather than a long  sentence
'`.Where(a => a.EventType == "leak" && a.TimeStamp.Month == DateTime.Now.Month)   

 .GroupBy(b => b.TimeStamp.Hour).ToList()`

Reply:
Done, please check